source/gen_security.rst

```diff
@@ -1,37 +1,371 @@
-========
+
+#########
 Security
-========
+#########
+
+Prime goals of ODAHU Security system is  to provide
+`authentication <https://en.wikipedia.org/wiki/Authentication>`_,
+`authorization <https://en.wikipedia.org/wiki/Authorization>`_
+and give users a flexible access control management solution.
+
+The first section `Security overview`_  shows the general design of authentication and authorization is described.
+Look at this section to have a deep understanding of how ODAHU security works under the hood or to learn basic concepts.
+
+The second section `Use built-in policies`_ describes how to add users permissions to use ODAHU via your idP_
+and ODAHU pre-defined roles
+
+The third section `Configure built-in policies`_ describes how you can get more flexibility
+such as adding new roles, map attributes for ODAHU RBAC from `JWT claims`_
+or writing your policy that could be more generic than RBAC like `Access based access control`_.
+
+
+
+.. contents:: Content table
+   :local:
+
+**************************
+Security overview
+**************************
+
+
+Component roles
+==================
+
+There are some common terms related to access control management systems.
+In this documentation, the next ones are commonly used.
+
+.. _idP:
+
+Identity Provider (idP)
+  A component that provides information about an entity (user or service).
+  In ODAHU the role of idP can do any `OpenID Connect`_ compatible provider.
+
+.. _PEP:
+
+Policy Enforcement Point (PEP)
+  A component that enforces security policies against each request to API or other protected resources.
+  In ODAHU the role of PEP plays `Envoy proxy`_.
+
+.. _PDP:
+
+Policy Decision Point (PDP)
+  A component that decides whether the request (action in the system) should be permitted or not.
+  In ODAHU role of PDP plays `OpenPolicyAgent`_.
```

Comment:
Shall we add this to the Glossary?

Reply:
Do you think it will be more convenient? I thought about it... But I am not sure that it simplifies documentation user experience

Reply:
I mean to leave here and add to the Glossary, just in case.